lib/chef/dsl/declare_resource.rb

```diff
@@ -155,7 +155,15 @@ def edit_resource!(type, name, created_at: nil, run_context: self.run_context, &
       def edit_resource(type, name, created_at: nil, run_context: self.run_context, &resource_attrs_block)
         edit_resource!(type, name, created_at: created_at, run_context: run_context, &resource_attrs_block)
       rescue Chef::Exceptions::ResourceNotFound
-        declare_resource(type, name, created_at: created_at, run_context: run_context, &resource_attrs_block)
+        resource = declare_resource(type, name, created_at: created_at, run_context: run_context)
+        if resource_attrs_block
+          if defined?(new_resource)
+            resource.instance_exec(new_resource, &resource_attrs_block)
```

Comment:
this was a bug that was getting papered over by the resource magic.  if you look at edit_resource! above it has all the same instance_exec stuff but if that throws then we catch here and never do the instance_exec fixup.  What was happening before was that you'd get property lookups through method_missing on the resource being constructed that bubbled up to the enclosing_provider which then hit the magic on the provider to forward the property to the resource.  So we broke that, and that exposed that this code had never gotten properly written.

Since if you remove this code it blows up the integration tests for accumulators written with edit_resource, we have coverage for it already.

TL;DR:  my head hurts.